kernel/src/process_standard.rs

```diff
@@ -1386,14 +1387,44 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
                         "Process not enabled flash={:#010X}-{:#010X} process={:?}",
                         app_flash.as_ptr() as usize,
                         app_flash.as_ptr() as usize + app_flash.len() - 1,
-                        process_name
+                        process_name.unwrap_or("(no name)")
                     );
                 }
             }
             // Return no process and the full memory slice we were given.
             return Ok((None, remaining_memory));
         }
 
+        if let Some((major, minor)) = tbf_header.get_kernel_version() {
+            // If the `KernelVersion` header is present, we read the requested kernel version and compare it to
+            // the running kernel version.
+            if crate::MAJOR != major || crate::MINOR < minor {
+                // If the kernel major version is different, we prevent the process from being loaded.
+                //
+                // If the kernel major version is the same, we compare the kernel minor version. The current
+                // running kernel minor version has to be greater or equal to the one that the process
+                // has requested. If not, we prevent the process from loading.
+                if config::CONFIG.debug_load_processes {
+                    debug!("WARN process {:?} not loaded as it requires kernel version >= {}.{} and < {}.0, (running kernel {}.{})", process_name.unwrap_or("(no name)"), major, minor, (major+1), crate::MAJOR, crate::MINOR);
+                }
+                return Err(ProcessLoadError::IncompatibleKernelVersion {
+                    version: Some((major, minor)),
+                });
+            }
+        } else {
+            if enforce_kernel_version {
```

Comment:
```suggestion
            if require_kernel_version {
```

I think the current logic makes sense: if a process tells us it requires a newer kernel version we always error in that case. If the process doesn't tell us what it requires we assume the process is fine, unless explicitly told not to. 

We may someday want to add a "force allow incompatible apps" or something, but we can make that change if the need ever arises.